src/main/java/com/google/cloud/genomics/dockerflow/task/TaskBuilder.java

```diff
@@ -163,38 +163,61 @@ public TaskBuilder inputArray(String name, String delimiter, String[] values) {
-   * container as ${name}.
-   *
-   * @param name
-   * @return
+   * Declare an input parameter for a file in GCS. The file will be accessible in the Docker
+   * container at the absolute local path given by ${name}.
    */
   public TaskBuilder inputFile(String name) {
-    return inputFile(name, null, null);
+    return inputPath(name, null, null, false);
   }
 
   /**
-   * Declare an input parameter for a local file. The file path will be accessible from the Docker
-   * container as ${name}.
-   *
-   * @param name
-   * @param localPath
-   * @return
+   * Declare and assign an input parameter for a file in GCS. The file will be accessible in
+   * the Docker container at the absolute local path given by ${name}.
    */
   public TaskBuilder inputFile(String name, String defaultGcsPath) {
-    return inputFile(name, defaultGcsPath, null);
+    return inputPath(name, defaultGcsPath, null, false);
   }
 
   /**
-   * Declare an input parameter for a local file and set a default. The file path will be accessible
-   * from the Docker container as ${name}.
-   *
-   * @param name
-   * @param defaultGcsPath
-   * @param localPath
-   * @return
+   * Declare and assign an input parameter for a file in GCS, explicitly providing a relative
+   * local path. The file will be accessible in the Docker container at the absolute local path
+   * given by ${name}.
    */
   public TaskBuilder inputFile(String name, String defaultGcsPath, String localPath) {
+	return inputPath(name, defaultGcsPath, localPath, false);
+  }
+
+  /**
+   * Declare and assign an input parameter for a folder in GCS. The folder will be accessible in
+   * the Docker container at the absolute local path given by ${name}.
+   */
+  public TaskBuilder inputFolder(String name) {
+    return inputPath(name, null, null, true);
+  }
+
+  /**
+   * Declare and assign an input parameter for a folder in GCS. The folder will be accessible in
+   * the Docker container at the absolute local path given by ${name}.
+   */
+  public TaskBuilder inputFolder(String name, String defaultGcsPath) {
+    return inputPath(name, defaultGcsPath, null, true);
+  }
+
+  /**
+   * Declare and assign an input parameter for a folder in GCS, explicitly providing a relative
+   * local path. The folder will be accessible in the Docker container at the absolute local path
+   * given by ${name}.
+   */
+  public TaskBuilder inputFolder(String name, String defaultGcsPath, String localPath) {
+	return inputPath(name, defaultGcsPath, localPath, true);
+  }
+ 
+  /**
+   * Declare and assign an input parameter for a path in GCS, explicitly providing a relative
+   * local path. The file will be accessible in the Docker container at the absolute local path
+   * given by ${name}. The path can be either a file or a folder.
+   */
+  private TaskBuilder inputPath(String name, String defaultGcsPath, String localPath, boolean isFolder) {
```

Comment:
In a few similar contexts, we've discussed that "localPath" is unclear.
We've called the GCS path the "remotePath" or "remoteURI" and the localPath the "dockerPath".